src/main/java/use_cases/login_leaderboard/RegisterUser.java

```diff
@@ -45,19 +47,29 @@ public void UserSetter(String username, String email, String password){
      * to the user data csv file.
      */
     @Override
-    public void createUser(){
-        if (isValid() && !UserAlreadyExists()){
+    public String createUser(){
+        if (isValid() && !UserAlreadyExists() && ValidEmail()){
             User current_user =  new User(this.username, this.email, this.password);
             // write to file
             // TODO
             userPresenter.PrepareView("You have been registered.");
+            return "yes";
+        }
+        if (isValid() && !UserAlreadyExists() && !ValidEmail()){
+            userPresenter.PrepareView("Enter a valid email.");
+            return "no";
         }
+
         else if(UserAlreadyExists()){
-            userPresenter.PrepareView("This user already exists");
+            userPresenter.PrepareView("This user already exists. Please log in now.");
+            return "yes";
+
         }
         else if(!isValid()){
             userPresenter.PrepareView("Your password is not valid");
+            return "no";
```

Comment:
You should have a text in the register panel the valid combination of characters/length to make a password valid. Otherwise the user would have to guess. Same goes for email

Reply:
How will the user login after registering if we can't go to the login panel from the register panel? Or back to the welcome screen?